packages/grafana-data/src/dataframe/DataFrameJSON.ts

```diff
@@ -188,16 +195,24 @@ export function dataFrameFromJSON(dto: DataFrameJSON): DataFrame {
       type = FieldType.string;
     }
 
+    const nanos = data?.nanos?.[index];
+
     // TODO: expand arrays further using bases,factors
 
-    return {
+    const dataFrameField: Field & { entities: FieldValueEntityLookup } = {
       ...f,
       type: type ?? guessFieldType(f.name, buffer),
       config: f.config ?? {},
       values: new ArrayVector(buffer),
       // the presence of this prop is an optimization signal & lookup for consumers
       entities: entities ?? {},
```

Comment:
nit: since you are not using `nanos` elsewhere it can be defined here inline: `nanos: data?.nanos?.[index],`

Reply:
yes, i agree. the question is this... the `nanos` field will always be there that way, but it's value will be `undefined`. in other words, with current code:
```js
{ a: 1, b: 2}
```

inline nanos code:
```js
{ a: 1, b: 2, nanos: undefined }
```

what i do not know, do we care about such details of not having a field "be there but value will be undefined"?